packages/navigation/src/hooks/useWebviewNavigate.ts

```diff
@@ -68,47 +84,34 @@ export function useWebviewNavigate<
         );
       }
 
-      if (typeof to !== 'string') {
+      if (to && typeof to !== 'string') {
+        // @ts-expect-error
         navigation.navigate(to.screen, to.params);
         return;
       }
 
       const { options } = linking;
 
       let path = to;
-      if (to.match(/^https?:\/\//)) {
-        path = extractPathFromURL(options?.prefixes, to) ?? '';
+      if (options?.prefixes && to.match(/^https?:\/\//)) {
+        path = extractPathFromURL(options.prefixes, to) ?? '';
       }
```

Comment:
Could be out-of-scope for this PR, but I find this confusing. We are stripping the host prefix, but only if the proposed URL contains `https://`? It feels like a wrong predicate here. I think it should more likely be `baseURL` that we checked for?

Reply:
As this is pre-existing logic - let's fix it in separate PR.